src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1679,10 +1680,19 @@ private function moveNode(string $srcAbsPath, string $destAbsPath): void
             throw new PathNotFoundException("Parent of the destination path '".$destAbsPath."' has to exist.");
         }
 
-        $query = 'SELECT path, id FROM phpcr_nodes WHERE path LIKE ? OR path = ? AND workspace_name = ? '.$this->getConnection()
-                ->getDatabasePlatform()->getForUpdateSQL()
+        $sql = $this->getConnection()->createQueryBuilder()
+            ->select('path, id')
+            ->from('phpcr_nodes')
+            ->where('(path LIKE :path_prefix OR path = :path) AND workspace_name = :workspace')
+            ->forUpdate()
```

Comment:
@derrabus did i understand your suggestion in https://github.com/jackalope/jackalope-doctrine-dbal/pull/436#discussion_r1470991804 wrong?

this breaks sqlite with "FOR UPDATE" not supported errors. or is that a bug in dbal and it is supposed to correctly decide if it should put FOR UPDATE into the statement?

Reply:
this still happens for sqlite, now that reverting mysql version and encoding got us rid of the fatal errors.

Reply:
okay, i see. in the context of the discussion (where we just got it from whichever platform was being used and got either nothing or the correct statement) i thought the query builder would be aware of the platform.

the unfortunate thing is that this makes the application think about specific implementations and have assumptions. it would fail with custom platform implementations or new implementations that also do not support the lock hint, or if some future sqlite version would support it...

thanks for the explanation, then i will hardcode the check for now.

Reply:
> i thought the query builder would be aware of the platform.

It is and this is why you get an exception. 🙃 

> the unfortunate thing is that this makes the application think about specific implementations and have assumptions. it would fail with custom platform implementations or new implementations that also do not support the lock hint, or if some future sqlite version would support it...

Yes, it would. I understand that this is a problem for you, hence the last paragraph of my previous comment. 🙂